src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
```

Comment:
It's assumed that the last element of the `fromFormat` array is the separator, but there's no validation checking if the actual separator in the `date` string matches the expected one from `fromFormat`. If the separator is different, the split operation will not work as intended.